src/DataBaseReadOnly.vala

```diff
@@ -621,71 +347,58 @@ public class FeedReader.DataBaseReadOnly : GLib.Object {
 
 	public Gee.HashMap<string, Article> read_article_stats(Gee.List<string> ids)
 	{
-		var query = new QueryBuilder(QueryType.SELECT, "articles");
-		query.selectField("articleID, unread, marked");
-		query.addRangeConditionString("articleID", ids);
-		query.build();
-
-		Sqlite.Statement stmt;
-		int ec = sqlite_db.prepare_v2(query.get(), query.get().length, out stmt);
-		if(ec != Sqlite.OK)
-		{
-			Logger.error(query.get());
-			Logger.error(sqlite_db.errmsg());
-		}
+		var query = "SELECT articleID, unread, marked FROM articles WHERE articleId IN (%s)".printf(n_params(ids.size));
+		var rows = m_db.executev(query, ids);
```

Comment:
This should be execute instead of executev

Reply:
In this one it's `executev` because `ids` is already a `List<string>`

Reply:
Ah alright, i just took a look using my phone and thought that it was a typo! should have been more careful :p

Reply:
At this point the commit is *very* broken, so I'm mainly looking for high-level comments. It's not even close to building yet :)